packages/api-client/src/hooks/useDarkModeState.ts

```diff
@@ -1,7 +1,7 @@
-import { ref, watch } from 'vue'
+import { onMounted, onUnmounted, ref, watch } from 'vue'
 
-// State
-const isDark = ref<boolean>(false)
+// State - null to indicate system preference
+const isDark = ref<boolean | null>(null)
```

Comment:
Might be nice to make this a union type instead, nullable booleans can be a little confusing
```suggestion
const colorScheme = ref<'light' | 'dark' | 'system'>('system')
```

Reply:
yes good call! i updated in the last commit in that sense 💪